includes/functions.php

```diff
@@ -1695,3 +1696,44 @@ function q_bridge_bits2indices($hex_data)
     }
     return $indices;
 }
+
+/**
+ * @param $device
+ * @param $raw_value
+ * @param $capacity
+ * @return float|int
+ */
+function get_toner_levels($device, $raw_value, $capacity)
+{
+    // -3 means some toner is left
+    if ($raw_value == '-3') {
+        return 50;
+    }
+
+    // -2 means unknown, -1 mean no restrictions
+    if ($raw_value == '-2' || $raw_value == '-1') {
+        return 0;  // FIXME: is 0 what we should return?
+    }
+
+    // corner cases
+    if ($device['os'] == 'ricoh' || $device['os'] == 'nrg' || $device['os'] == 'lanier') {
+        if ($raw_value == '-100') {
+            return 0;
+        }
+    } elseif ($device['os'] == 'brother') {
+        if (!str_contains($device['hardware'], 'NC-8600h')) {
```

Comment:
please note that the hardware value of brother nc-8600h device is something like MFC-L8850